File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> seen;
        for (int i : nums) {
            if (!seen[i]) {
                seen[i] = 1;
            } else if (seen[i] == 1) {
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        seen = dict()
        for i, num in enumerate(numbers):
            r = target - num
            if r in seen:
                return [seen[r]+1, i+1]
            else:
                seen[num] = i